File: example/typescript/test/OtherModule/Fixtures/Builders.ts
namespace OtherModule.Builder {
    export interface OtherPropertyDef {
        id?: string,
        name?: string,
    }
    export function otherProperty(def?: OtherPropertyDef): OtherProperty {
        const id = def?.id ?? "someValue"
        const name = def?.name ?? "someValue"

        return OtherProperty.create(
            id,
            name,
        )
    }

    export interface OtherClassDef {
        id?: string,
        amount?: number,
    }
    export function otherClass(def?: OtherClassDef): OtherClass {
        const id = def?.id ?? "someValue"
        const amount = def?.amount ?? 0

        return new OtherClass(
            new OtherId(id),
            amount,
        )
    }
}

File: example/kotlin/src/testFixtures/kotlin/com/some/pkg/othermodule/Builders.kt
package com.some.pkg.othermodule

import com.some.pkg.othermodule.api.OtherClass
import com.some.pkg.othermodule.api.OtherId

data class OtherClassDef(
    var id: String = "someValue",
    var amount: Int = 0,
)
fun otherClass(init: OtherClassDef.() -> Unit = {}): OtherClass {
    val def = OtherClassDef().apply(init)
    return OtherClass(
        id = OtherId(def.id),
        amount = def.amount,
    )
}